doc/internals/variable-objects.rst

```diff
@@ -1,31 +0,0 @@
-Variable objects
```

Comment:
All of this content has been kept in `internal-design.rst`. I'm not sure why `git mv` didn't seem to understand that I was just renaming this file.